src/Stratis.Bitcoin.Features.BlockStore/BlockStoreCache.cs

```diff
@@ -34,15 +34,42 @@ public class BlockStoreCache : IBlockStoreCache
         /// <summary>Provider of time functions.</summary>
         protected readonly IDateTimeProvider dateTimeProvider;
 
+        public readonly int MaxCacheBlocksCount;
+
+        /// <summary>Entry options for adding blocks to the cache.</summary>
+        private readonly MemoryCacheEntryOptions blockEntryOptions;
+
+        /// <summary>Entry options for adding transactions to the cache.</summary>
+        private readonly MemoryCacheEntryOptions txEntryOptions;
+
         public BlockStoreCache(IBlockRepository blockRepository, IDateTimeProvider dateTimeProvider, ILoggerFactory loggerFactory)
-            : this(blockRepository, new MemoryCache(new MemoryCacheOptions()), dateTimeProvider, loggerFactory)
+            : this(blockRepository, dateTimeProvider, loggerFactory, null)
         {
         }
 
-        public BlockStoreCache(IBlockRepository blockRepository, IMemoryCache memoryCache, IDateTimeProvider dateTimeProvider, ILoggerFactory loggerFactory)
+        public BlockStoreCache(IBlockRepository blockRepository, IDateTimeProvider dateTimeProvider, ILoggerFactory loggerFactory, IMemoryCache memoryCache = null)
```

Comment:
It appears that either the other constructor is not required or this `= null` is not required.

Reply:
You are right. 
I thought that first constructor was used by the DI but after that '= null' it's no longer true. Thanks for pointing this out!